app/models/user.rb

```diff
@@ -1251,10 +1251,21 @@ def serializable_hash(options = nil)
     # NOTE: doing deep_dup is necessary here to avoid changing the inner values
     # of the freezed variables (which would leak PII)!
     default_options = DEFAULT_SERIALIZE_OPTIONS.deep_dup
+
+    include_email = options&.[](:include_email)
+    exclude_deprecated = options&.[](:exclude_deprecated)
+
     # Delegates's emails and regions are public information.
-    default_options[:methods].push("email", "location", "region_id") if staff_delegate?
+    default_options[:methods].push("email") if include_email || staff_delegate?
+    default_options[:methods].push("location", "region_id") if !exclude_deprecated && staff_delegate?
 
     options = default_options.merge(options || {}).deep_dup
+
+    if exclude_deprecated
+      options[:methods].delete("delegate_status")
+      options[:include].delete("teams")
+    end
```

Comment:
This feels a bit redundant with your other code change above, because you're only adding `delegate_status` and `teams` if the condition `if !exclude_deprecated` is already met. Why are you running this here again?